### PR TITLE
[Plugin] stylish-haskell formatter

### PR DESCRIPTION
**Pull request by Ailrun**
Is it OK to use "stylishHaskellCmd:" error message when the name of the plugin is "stylish-haskell"? All existing plugins have names without "-", so I couldn't find an example for this kind of naming.

Done:
- [x] Add stylish-haskell plugin
- [x] Add some tests for stylish-haskell formatting

### CODE REVIEW

**Comment by Ailrun** [comment]
@fendor Thank you for the review and answer, now I added tests and believe it's ready for more reviews and merge.

File: test/functional/Format.hs [hunk 1]
```diff
@@ -68,6 +69,38 @@ providerTests = testGroup "formatting provider" [
...
+        "import           Data.Char\n\
+        \import qualified Data.List\n\
+        \import           Data.String\n\
+        \\n\
+        \bar :: Maybe (Either String Integer) -> Integer\n\
+        \bar Nothing = 0\n\
+        \bar (Just (Left _)) = 0\n\
+        \bar (Just (Right x)) = x\n"
+  ]
+
```
Comment (Ailrun) [inline test/functional/Format.hs:103]: While I was writing this, I thought it might be better to add a golden test facility.
Any opinions/ideas?

Comment (fendor) [reply test/functional/Format.hs:103]: Good idea, should be easy to integrate with https://hackage.haskell.org/package/tasty-golden

Comment (Ailrun) [reply test/functional/Format.hs:103]: I will try that in a separate PR. Thank you for the suggestion :)

**Pull request merged**